pkg/util/proto/testing/openapi.go

```diff
@@ -32,6 +34,10 @@ type Fake struct {
 	once     sync.Once
 	document *openapi_v2.Document
 	err      error
+
+	v3DocumentsLock sync.Mutex
```

Comment:
You have members of that struct that only make sense for v3 and members that only make sense for v2. Can you split into two different structs?

Reply:
yeah fair enough